guides/source/action_cable_overview.md

```diff
@@ -396,46 +421,73 @@ appear/disappear API could be backed by Redis, a database, or whatever else.
 
 Create the client-side appearance channel subscription:
 
-```coffeescript
-# app/assets/javascripts/cable/subscriptions/appearance.coffee
-App.cable.subscriptions.create "AppearanceChannel",
-  # Called when the subscription is ready for use on the server.
-  connected: ->
-    @install()
-    @appear()
-
-  # Called when the WebSocket connection is closed.
-  disconnected: ->
-    @uninstall()
-
-  # Called when the subscription is rejected by the server.
-  rejected: ->
-    @uninstall()
-
-  appear: ->
-    # Calls `AppearanceChannel#appear(data)` on the server.
-    @perform("appear", appearing_on: $("main").data("appearing-on"))
-
-  away: ->
-    # Calls `AppearanceChannel#away` on the server.
-    @perform("away")
-
-
-  buttonSelector = "[data-behavior~=appear_away]"
-
-  install: ->
-    $(document).on "turbolinks:load.appearance", =>
-      @appear()
-
-    $(document).on "click.appearance", buttonSelector, =>
-      @away()
-      false
-
-    $(buttonSelector).show()
-
-  uninstall: ->
-    $(document).off(".appearance")
-    $(buttonSelector).hide()
+```js
+// app/javascript/cable/subscriptions/appearance_channel.js
+import consumer from "./consumer"
+
+const onLoadAppearance  = "turbolinks:load.appearance"
+const onClickAppearance = "click.appearance"
+const buttonSelector    = "[data-behavior~=appear_away]"
+
+consumer.subscriptions.create("AppearanceChannel", {
+  // Called when the subscription is ready for use on the server.
+  connected() {
+    this.install()
+    this.appear()
+  },
+
+  // Called when the WebSocket connection is closed.
+  disconnected() {
+    this.uninstall()
+  },
+
+  // Called when the subscription is rejected by the server.
+  rejected() {
+    this.uninstall()
+  },
+
+  appear() {
+    const main = document.querySelector("main")
+    const appearingOn = main.getAttribute("data-appearing-on")
+    // Calls `AppearanceChannel#appear(data)` on the server.
+    this.perform("appear", { appearing_on: appearingOn })
+  },
+
+  away() {
+    // Calls `AppearanceChannel#away` on the server.
+    this.perform("away")
+  },
+
+  install() {
+    document.addEventListener(onLoadAppearance, () => this.appear())
+    document.addEventListener(onClickAppearance,
+      (element) => {
+        if (element.matches(buttonSelector)) {
+          this.away()
+        }
+        return false
+      }
+    )
+
+    document.querySelectorAll(buttonSelector).forEach(
+      (element) => element.style.display = "block"
+    )
+  },
+
+  uninstall() {
+    const listeners = getEventListeners(document)
+    listeners[onLoadAppearance].forEach(
+      (listener) => document.removeEventListener(onLoadAppearance, listener.listener)
+    )
+    listeners[onClickAppearance].forEach(
+      (listener) => document.removeEventListener(onClickAppearance, listener.listener)
+    )
+
+    document.querySelectorAll(buttonSelector).forEach(
+      (element) => element.style.display = "none"
+    )
+  }
+})
```

Comment:
Unrelated to your conversion, this example could use some freshening up. Here's a more realistic version extracted from Basecamp:
<details>

```js
// app/javascript/cable/subscriptions/appearance_channel.js
import consumer from "./consumer"

consumer.subscriptions.create("AppearanceChannel", {
  // Called once when the subscription is created.
  initialized() {
    this.update = this.update.bind(this)
  },
  
  // Called when the subscription is ready for use on the server.
  connected() {
    this.install()
    this.update()
  },

  // Called when the WebSocket connection is closed.
  disconnected() {
    this.uninstall()
  },

  // Called when the subscription is rejected by the server.
  rejected() {
    this.uninstall()
  },

  update() {
    this.documentIsActive ? this.appear() : this.away()
  },

  appear() {
    // Calls `AppearanceChannel#appear(data)` on the server.
    this.perform("appear", { appearing_on: this.appearingOn })
  },

  away() {
    // Calls `AppearanceChannel#away` on the server.
    this.perform("away")
  },

  install() {
    window.addEventListener("focus", this.update)
    window.addEventListener("blur", this.update)
    document.addEventListener("turbolinks:load", this.update)
    document.addEventListener("visibilitychange", this.update)
  },

  uninstall() {
    window.removeEventListener("focus", this.update)
    window.removeEventListener("blur", this.update)
    document.removeEventListener("turbolinks:load", this.update)
    document.removeEventListener("visibilitychange", this.update)
  },

  get documentIsActive() {
    return document.visibilityState == "visible" && document.hasFocus()
  },
  
  get appearingOn() {
    const element = document.querySelector("[data-appearing-on]")
    return element ? element.getAttribute("data-appearing-on") : null
  }
})
```

</details>

Reply:
@javan should we consider the people reading this guide to know JS well? Because I was definitely confused by this `bind`. It's fine example of a quirk though.

```js
initialized() {
  this.update = this.update.bind(this)
}
```

I'm also a little curious as to what you store in the `data-appearing-on` attribute?

Reply:
I'm not sure what the target experience level is for this guide. `bind`ing that function allows the event listeners to be installed and uninstalled correctly, something that was done in the previous iteration using [jQuery's "namespaced" event syntax](http://api.jquery.com/on/#event-names), which I also find confusing and require some level of jQuery expertise to grok.

<img width="524" alt="screen shot 2019-01-03 at 10 48 08 am" src="https://user-images.githubusercontent.com/5355/50647247-17a00080-0f46-11e9-852d-ff1ad34ab9f8.png">

Reply:
I believe we store a GlobalID in `data-appearing-on` corresponding to the primary DB record on the page, e.g. a `Todo` or `Message`.

Reply:
True, that jQuery isn't exactly easier to understand. Let's just keep the `bind` and see how it goes 👍